### ISSUE TITLE
Cleanup File Locations Appending _1 to file names

### ISSUE THREAD

**Issue by agent462** [xLightsSequencer/xLights#4970]
**Describe the bug**
I am migrating 2023 sequences to 2024 sequences.  When I open a 2023 sequence and save it into the 2024 folder, run a check sequence it obviously shows the file locations in 2023.   When I click cleanup file locations it will fix the folder but append _1 to file names breaking the link to them.  When you run check sequence again it will show all the missing files with the correct folder but all having _1.   The files already exist in the 2024 location.

**To Reproduce**
(and/or link to a short video showing the problem with audio track describing what you are doing)
1. Save sequence from an old show folder to a new show folder
2. Run check sequence and see file names are proper but not show folder
3. Run cleanup file locations
4. Run check sequence again and see that folders are correct but filenames are wrong

**Expected behavior**
Just fix the file folder structure. 

**Screenshots**
<img width="1162" alt="Screenshot 2024-11-18 at 9 11 08 PM" src="https://github.com/user-attachments/assets/7df0a5f3-8ad9-41ca-8d78-79e26d8eed83">

<img width="1237" alt="Screenshot 2024-11-18 at 9 11 41 PM" src="https://github.com/user-attachments/assets/2dd783e6-9aaf-4b89-999f-65708c879f7c">

**Versions (please complete the following information):**
 - OS: OSX 15.1
 - xLights version 2024.18

**Additional context**
I have seen this on a lot of sequences, not just one.

**Comment by cybercop23** [comment]
That's what Clean File Location does. Takes the item from the original location, renames it by adding 1 (or 2 - next number if already exists) and makes a copy. Have you saved the sequence after you ran cleanup file location and then reopen it again? You can also click off the effect and back on it and it should point to the new location.

**Comment by agent462** [comment]
Yes I have but it doesn't actually copy the files.  The _1 files do not exist.

**Comment by cybercop23** [comment]
Something must be off with your read/write. Added Image effect from an outside folder.. 2023 show.. ran CFL and it made a copy ( no_1 since one didn't aleady exist) and it was changed.
![image](https://github.com/user-attachments/assets/a34159d2-d5b0-4caa-9f96-b18b06757060)

Do you have any errors or info in Tools -> View Log?

**Comment by agent462** [comment]
No errors but I can see it attempting to copy the files but it reuses shaders (I guess this is what I was expecting for other files.). I've given xLights full disk access - I can try to play around with other permissions.

2024-11-18 23:14:39,452 8291792960 log_base [DEBUG] Cleaning up file locations.
2024-11-18 23:14:39,456 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,457 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,457 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,457 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,458 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,460 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/Mega Tree_Everybody Loves Xmas.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/Mega Tree_Everybody Loves Xmas_1.mp4.
2024-11-18 23:14:39,463 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Elf.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Elf_1.mp4.
2024-11-18 23:14:39,465 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Ginger w Hat.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Ginger w Hat_1.mp4.
2024-11-18 23:14:39,467 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Claus.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Claus_1.mp4.
2024-11-18 23:14:39,467 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,468 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,469 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Baby doll_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/Baby doll_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,471 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Illustrate a scene where Momma gently tells Little Johnny that he's too young. Include the detail in_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/Illustrate a scene where Momma gently tells Little Johnny that he's too young. Include the detail in_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,471 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,473 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/rocking horse_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/rocking horse_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,474 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/toy BB gun_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/toy BB gun_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,476 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Baby doll_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/Baby doll_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,477 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Illustrate a scene where Momma gently tells Little Johnny that he's too young. Include the detail in_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/Illustrate a scene where Momma gently tells Little Johnny that he's too young. Include the detail in_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,478 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Elf.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Elf_1.mp4.
2024-11-18 23:14:39,479 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Ginger w Hat.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Ginger w Hat_1.mp4.
2024-11-18 23:14:39,480 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Videos/_Dancing Claus.mp4 to /Users/bryan/Desktop/xlights/2024/Videos/_Dancing Claus_1.mp4.
2024-11-18 23:14:39,480 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,480 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,481 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Christmas Tree.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Christmas Tree.fs.
2024-11-18 23:14:39,482 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Christmas Present.png to /Users/bryan/Desktop/xlights/2024/Images/Christmas Present_1.png.
2024-11-18 23:14:39,483 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Christmas Present2.png to /Users/bryan/Desktop/xlights/2024/Images/Christmas Present2_1.png.
2024-11-18 23:14:39,484 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Christmas Present3.png to /Users/bryan/Desktop/xlights/2024/Images/Christmas Present3_1.png.
2024-11-18 23:14:39,485 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/Christmas Present4.png to /Users/bryan/Desktop/xlights/2024/Images/Christmas Present4_1.png.
2024-11-18 23:14:39,485 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Christmas Tree.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Christmas Tree.fs.
2024-11-18 23:14:39,486 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/rocking horse_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/rocking horse_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,487 8291792960 log_base [DEBUG] Copying file /Users/bryan/Desktop/xlights/2023/sequences/Images/toy BB gun_clipdrop-background-removal.png to /Users/bryan/Desktop/xlights/2024/Images/toy BB gun_clipdrop-background-removal_1.png.
2024-11-18 23:14:39,487 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.
2024-11-18 23:14:39,487 8291792960 log_base [DEBUG] Reusing file /Users/bryan/Desktop/xlights/2024/Shaders/Creation by Silexars.fs for /Users/bryan/Desktop/xlights/2023/Shaders/Creation by Silexars.fs.

<img width="482" alt="Screenshot 2024-11-18 at 11 18 38 PM" src="https://github.com/user-attachments/assets/7b1eb580-8de9-4912-a2c3-04386fdfa693">

**Comment by cybercop23** [comment]
Ahhh.. I missed that... Mac... yes, for shaders it reuses names since it is unlikely to have a name collision. NOt so much for images and videos. Okay.. try to restart mac after giving it permissions and see.

**Comment by agent462** [comment]
Same issue on reboot.  Let me move the show directory off the Desktop.  Maybe the new permission model in OSX is still screwing with it.

**Comment by agent462** [comment]
Same issue in a new show directory off my user folder.  I will keep looking at xlights permissions to disk but it can save sequences just fine and everything.

Sequence effect problems

    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 0:12.275
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 0:56.150
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 1:55.800
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 2:10.200
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 0:04.275
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 0:19.175
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 0:50.675
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 1:34.525
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 1:49.425
    ERR: Video effect video file '/Users/bryan/xlights/2024/Videos/108_2.mp4' does not exist. Model 'Column Matrix GRP', Start 2:03.425

**Comment by agent462** [comment]
I also tried uninstalling XLights and Reinstalling.

OK Yes OSX is blocking this.  This is from the OSX Console Log:
error	23:43:55.266876-0600	kernel	Sandbox: xLights(2870) deny(1) file-read-data /Users/bryan/Desktop/xlights/2022/sequences/Videos/108.mp4

**Comment by computergeek1507** [comment]
It should only add an underscore 1,2 if the files exists in the new location, if it is added it to the names for no reason, id consider it a bug

**Comment by derwin12** [comment]
Try xlights->preferences .. i think sequence where there is a media section - add that directory to that and mac will get the perms it needs. I think.

**Comment by agent462** [comment]
Just an update.  

If I add the directories in xlights->preferences xlights no longer complains about the files being somewhere else.  When cleaning up file locations though it doesn't references the files to the new show folder.  So basically it's just pointing at the old location.

After running cleanup:
<img width="1349" alt="Screenshot 2024-11-19 at 7 41 07 PM" src="https://github.com/user-attachments/assets/3477955f-2a24-414f-bf4a-ccdcaa67237c">

Even with "full disk access", I can't get OSX to give xlights permissions to these folders.  I honestly am not familiar with the xlights code or build process but googling the OSX console error you get a lot of references to checking NO in "User Script Sandboxing" for build settings.

**Comment by agent462** [comment]
<img width="475" alt="Screenshot 2024-11-19 at 7 55 53 PM" src="https://github.com/user-attachments/assets/bb66d198-cb0c-4eeb-bfbc-d0889e0efde7">

**Comment by agent462** [comment]
Success!  Moving my xlights folder to ~/Documents/xlights/ allows xlights to do what it needs.  

I think there is still a story or two worthy here.  
1) Better log handling when OSX gets file permission errors (it was only found in the OSX console log
2) Get Xlights to have OSX prompt for the disk access it needs. I referenced the "User Script Sandboxing"  setting above and I don't know whether it's relevant or not.

Should I close this and create a new "issue"?

**Comment by cybercop23** [comment]
Yes, please close and try to recreate on a new empty folder and provide steps to reproduce.
